velox/expression/tests/ExpressionVerifier.cpp

```diff
@@ -121,25 +139,47 @@ void compareVectors(const VectorPtr& left, const VectorPtr& right) {
   LOG(INFO) << "All results match.";
 }
 
+// Returns a copy of 'rowVector' but with the columns having indices listed in
+// 'columnsToWarpInLazy' wrapped in lazy encoding. 'columnsToWarpInLazy' should
+// be a sorted list of column indices.
+RowVectorPtr wrapColumnsInLazy(
```

Comment:
Can we somehow reuse `VectorFuzzer::fuzzRowChildrenToLazy()` here? (say we push the coin toss logic down to VectorFuzzer).

Reply:
fuzzRowChildrenToLazy() randomly selects columns to fuzz with lazy, however in this case we need to retain the information about which columns would be fuzzed so that we can recreate the same set of columns for when the test re-runs after applying TRY() (when in the first try both common and simplified path return the same error). Moreover, this also helps serialize the list for quick repro using vector fuzzer

Reply:
> however in this case we need to retain the information about which columns would be fuzzed

could we achieve the same by calling the VectorFuzzer method above, then iterating over the generated columns collecting which ones have `isLazy()` set?

Reply:
That was my first attempt too, but the problem then arises is when we need to retain that information when we re-run the same expression but with TRY() added (https://github.com/facebookincubator/velox/blob/043e397822cdbf38b42717634fec127879e42d39/velox/expression/tests/ExpressionFuzzer.cpp#L730-L743), this is where we need to retain the information about which columns were fuzzed and have a utility function wrapColumnsInLazy() to apply it. If I instead use fuzzRowChildrenToLazy(), I would have to add that code in ExpressionFuzzer.cpp that extracts the info about fuzzer column idxs, then pass 2 different row vectors (Since lazy vectors cannot be copied, I'll need a utility to apply the same wrapping again when re-running with TRY). Basically the amount of code would be the same, just some of it would moved around between ExpressionFuzzer and ExpressionVerifier

Reply:
I see, it makes more sense now. Thanks for explaining it!